monitoring/mongodb/alerts.yaml

```
@@ -183,10 +183,10 @@ groups:
 
   - alert: MongoDbRSNotSynced
     expr: |
-      sum by (rs_nm) (mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*", member_state="SECONDARY"}) != (${replicas} - 1)
+      count by (rs_nm, pod) (mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*", member_state="SECONDARY", rs_state="1"}) != (${replicas} - 1)
```

The pod was added , as when we don't add this additional filtering we end up with a total of 3 instead of 2 , this is explained by the fact that for each pod we have 2 metrics exported , given the fact that these metrics actually have the same pos, namespace, rs_state and state member it is concidered as 1 metric so gives 1 x 3 which results in 3 , when filtering by the pod as well we make sure that for each pod the exported metrics return indeed 2 secondaries. please don't hesitate to challenge this

---

* why do we filter on `rs_state=1` : are the wrong states not covered by another alert, or is this covering extra case?
* if there is an issue, this will however trigger multiple alerts: typically, if a single (mongo) pod is down, each of the remaining pod will generate an alert, and 2 alerts will be raised instead of 1 → should avoid this duplication
* the `pod` is actually not relevant -and possibly misleading- as a parameter of the alert: as it seems to indicate an issue with this pod....when actually the issue is with the "remaining" pod: the one(s) which may not actually trigger the alert... → need to make sure the `pod` label does not end up in the final time serie

maybe something like this would solve the issue:

```suggestion
      group by(rs_nm)(
         count by (rs_nm, pod) (mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*", member_state="SECONDARY"}) != (${replicas} - 1)
      )
```